documentation/AuthorizationService.md

```diff
@@ -1,123 +1,245 @@
 * **parameters:** 
-    * userId: Number. A system generated unique number that identifies the user.
-    * clientId: Number. A system generated unique number that identifies the client.
-    * redirectUri: String. The client redirection URI that the authorization server directs the user-agent to when a authorization decision is established.
-    * selectedPreferences: Object. An object specifying the preferences that the user has selected to share, in the privacy ontology. An example:
+    * gpiiToken: String. A GPII token that associates with user preferences.
+    * clientId: String. A system generated unique string that identifies the client.
+* **return:** Object. Contains an access token and the number of seconds that the access token will expire. For example:
+```
+{
+    "accessToken": "8ea3457bf283db5d34ea5a4079fa36b2",
+    "expiresIn": 3600
+}
+```
+Return an object that contains the error message and http status code if,
++ The GPII token is wrong, or,
++ The client is not a GPII App Installation Client.
+
+#### grantPrivilegedPrefsCreatorAuthorization(clientId, scope)
+* **description**: Grant an authorization to a privileged preferences creator. This authorization allows a privileged preferences creator to add new preferences sets. The scenarios handled by this function: 
+    * If the given privileged preferences creator has never been assigned an authorization, the function will generate and return one.
+    * If the given privileged preferences creator has already been assigned an authorization,the function will return this existing authorization.
+* **parameters:** 
+    * clientId: String. A system generated unique number that identifies the client.
+        - Notes: "privilegedPrefsCreatorClient" is the only GPII client type that is allowed to request privileged prefs creator authorizations. See [the `Privileged Prefs Creator Clients` document structure](AuthServer.md#privileged-prefs-creator-clients).
+    * scope: String. Must be "add_preferences".
+        - Notes: "add_preferences" is the only scope currently supported by the [GPII Client Credentials Grant](https://wiki.gpii.net/w/GPII_OAuth_2_Guide#Client_Credentials_Grant).
+* **return:** String. A privileged prefs creator access token. Return an object that contains the error message and http status code if the scope is wrong, or the client is not allowed to add preferences, or the client type is not "privilegedPrefsCreatorClient".
+
+#### grantWebPrefsConsumerAuthCode(userId, clientId, redirectUri, selectedPreferences)
+- **description**: Grant an Authorization Code for the specified user, client, redirect URI and selected preferences. We first check to see if we have an existing authorization for the user, client, and redirect URI. If we do, we issue a new code for that authorization. Otherwise we create a new authorization record and a new code.
+- **parameters:** 
+    - userId: String. A system generated unique number that identifies the user.
```

Comment:
Are these actually numbers?

Reply:
Right now GPII doesn't have the feature for creating users. When we have this feature, I believe `userId`s will be either GPII internally generated ids or [CouchDB generated IDs](http://guide.couchdb.org/draft/tour.html#first). Both likely are UUIDs, which are strings. For now, the mock data uses strings such as "user-1" for this field value.

Reply:
We should rename all these uses of "numbers" to strings

Reply:
Renamed "numbers" to "ids". More general regardless the value type. :)